resources/options.js

```diff
@@ -76,50 +83,75 @@ function restoreOptionsAvailability() {
 
 // Shows the sub-options.
 function showSubOptions() {
-    $('#security_sub').show();
-    $('#privacy_sub').show();
-    $('#hints_sub').show();
+    for(var i = 0; i < subOptions.length; i++) {
+        $(subOptionsID(i)).show();
+    }
 }
 
 // Hides the sub-options.
 function hideSubOptions() {
-    $('#security_sub').hide();
-    $('#privacy_sub').hide();
-    $('#hints_sub').hide();
+    for(var i = 0; i < subOptions.length; i++) {
+        $(subOptionsID(i)).hide();
+    }
 }
 
 // Shows the option cards.
 function showCards() {
-    $('#security').show();
-    $('#privacy').show();
-    $('#hints').show();
+    for(var i = 0; i < cards.length; i++) {
+        $(cardID(i)).show();
+    }
 }
 
 // Hides the option cards.
 function hideCards() {
-    $('#security').hide();
-    $('#privacy').hide();
-    $('#hints').hide();
+    for(var i = 0; i < cards.length; i++) {
+        $(cardID(i)).hide();
+    }
 }
 
 // Switches availability of options.
 function switchDisable() {
-    $('#tabs').disabled = !$('#tabs').disabled;
-    $('#comments').disabled = !$('#comments').disabled;
-    $('#peerComments').disabled = !$('#peerComments').disabled;
-    $('#focus').disabled = !$('#focus').disabled;
-    $('#username').disabled = !$('#username').disabled;
-    $('#repo').disabled = !$('#repo').disabled;
-    $('#file').disabled = !$('#file').disabled;
+    for(var i = 1; i < options.length; i++) {
+        $(optionsID(i)).disabled = !$(optionsID(i)).disabled;
+    }
 }
 
 // Restores availability of options.
 // The availability depends on the value of the logging checkbox state.
 function restoreDisable() {
-    $('#tabs').disabled = $('#comments').disabled =
-        $('#peerComments').disabled = $('#focus').disabled =
-            $('#username').disabled = $('#repo').disabled =
-                $('#file').disabled = !items.loggingEnabled;
+    var enabled = !items.loggingEnabled;
+    for(var i = 1; i < options.length; i++) {
+        $(optionsID(i)).disabled = enabled;
+    }
+}
+
+// Retrieves the ID of an option.
+function optionsID(index){
+    return idGenerator(index, options);
+}
+
+// Retrieves the ID of a subOption.
+function subOptionsID(index){
+    return idGenerator(index, subOptions);
+}
+
+// Retrieves the ID of a card.
+function cardID(index){
+    return idGenerator(index, cards);
+}
+
+// Helper function that appends a hashtag to an element from an array, in order to return an ID.
+function idGenerator(index, array){
+    return '#' + array[index];
+}
```

Comment:
Good idea, will do!
I will then also rename the function, as it's not an ID generator anymore. ;)